e2e-tests/src/test/electing_validators.rs

```diff
@@ -0,0 +1,266 @@
+    wait_for_full_era_completion, wait_for_session, AnyConnection, KeyPair, RootConnection,
+    SignedConnection,
+};
+use log::info;
+use pallet_elections::CommitteeSeats;
+use primitives::{staking::MIN_VALIDATOR_BOND, EraIndex, TOKEN};
+use sp_core::{storage::StorageKey, Pair};
+use substrate_api_client::{compose_extrinsic, AccountId, XtStatus};
+
+use crate::{accounts::get_sudo_key, Config};
+
+/// Gathers keys and accounts for all validators used in the experiment.
+struct Accounts {
+    stash_keys: Vec<KeyPair>,
+    stash_accounts: Vec<AccountId>,
+
+    controller_keys: Vec<KeyPair>,
+    controller_accounts: Vec<AccountId>,
+}
+
+/// Generate `Accounts` struct.
+fn setup_accounts() -> Accounts {
+    let seeds = (0..6).map(|idx| format!("//Validator//{}", idx));
+
+    let stash_seeds = seeds.clone().map(|seed| format!("{}//Stash", seed));
+    let stash_keys = stash_seeds.map(|s| keypair_from_string(&s));
+    let stash_accounts = stash_keys.clone().map(|k| account_from_keypair(&k));
+
+    let controller_seeds = seeds.map(|seed| format!("{}//Controller", seed));
+    let controller_keys = controller_seeds.map(|s| keypair_from_string(&s));
+    let controller_accounts = controller_keys.clone().map(|k| account_from_keypair(&k));
+
+    Accounts {
+        stash_keys: stash_keys.collect(),
+        stash_accounts: stash_accounts.collect(),
+        controller_keys: controller_keys.collect(),
+        controller_accounts: controller_accounts.collect(),
+    }
+}
+
+/// Endow validators (stashes and controllers), bond and rotate keys.
+fn prepare_validators(root_connection: &RootConnection, node: &str, accounts: &Accounts) {
+    balances_batch_transfer(
+        &root_connection.as_signed(),
+        accounts.stash_accounts.clone(),
+        MIN_VALIDATOR_BOND + TOKEN,
+    );
+    balances_batch_transfer(
+        &root_connection.as_signed(),
+        accounts.controller_accounts.clone(),
+        TOKEN,
+    );
+
+    for (stash, controller) in accounts
+        .stash_keys
+        .iter()
+        .zip(accounts.controller_accounts.iter())
+    {
+        let connection = SignedConnection::new(node, stash.clone());
+        staking_bond(
+            &connection,
+            MIN_VALIDATOR_BOND,
+            controller,
+            XtStatus::Finalized,
+        );
+    }
+
+    for controller in accounts.controller_keys.iter() {
+        let keys = rotate_keys(root_connection).expect("Failed to retrieve keys from chain");
+        let connection = SignedConnection::new(node, controller.clone());
+        set_keys(&connection, keys, XtStatus::Finalized);
+        staking_validate(&connection, 10, XtStatus::Finalized);
+    }
+}
+
+/// Produce storage key to `ErasStakers::era`.
+///
+/// Since in `substrate-api-client` it seems impossible to get prefix for the first key in double
+/// map, we have to do it by hand.
+fn get_eras_stakers_storage_key(era: EraIndex) -> StorageKey {
```

Comment:
what about method `get_storage_map_key_prefix`

Reply:
this works with Map, not DoubleMap, so it won't work since corresponding structure will not be found in metadata